<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />



<title>Overview of Data Retrieval Workflow</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Overview of Data Retrieval Workflow</h1>



<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(cbioportalR)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>We will outline the main data retrieval workflow and functions using
a case study based on two public sets of data:</p>
<ol style="list-style-type: decimal">
<li>105 samples in high risk nonmuscle invasive bladder cancer patients
<a href="https://pubmed.ncbi.nlm.nih.gov/28583311/">(Pietzak et
al. 2017)</a>.</li>
<li>18 samples of 18 prostate cancer patients <a href="https://pubmed.ncbi.nlm.nih.gov/31564440/">(Granlund et
al. 2020)</a></li>
</ol>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<p>Before accessing data you will need to connect to a cBioPortal
database and set your base URL for the R session. In this example we
will use data from the public cBioPortal database instance (<a href="https://www.cbioportal.org" class="uri">https://www.cbioportal.org</a>). You do not need a token to
access this public website. If you are using a private instance of
cBioPortal (like MSK’s institutional database), you will need to acquire
a token and save it to your .Renviron file.</p>
<p><em>Note: If you are a MSK researcher working on IMPACT, you should
connect to MSK’s cBioPortal instance to get the most up to date IMPACT
data, and you must follow MSK-IMPACT publication guidelines when using
the data.</em></p>
<p>To set the database url for your current R session use the
<code>set_cbioportal_db()</code> function. To set it to the public
instance you can either provide the full URL to the function, or just
<code>public</code> as a shortcut. This function will both check your
connection to the database and set the url
(<code>www.cbioportal.org/api</code>) as your base url to connect to for
all future API calls during your session.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">set_cbioportal_db</span>(<span class="st">&quot;public&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="co">#&gt; ✔ You are successfully connected!</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a><span class="co">#&gt; ✔ base_url for this R session is now set to &quot;www.cbioportal.org/api&quot;</span></span></code></pre></div>
<p>You can use <code>test_cbioportal_db</code> at any time throughout
your session to check your connection. This can be helpful when
troubleshooting issues with your API calls.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">test_cbioportal_db</span>()</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co">#&gt; ✔ You are successfully connected!</span></span></code></pre></div>
</div>
<div id="get-study-metadata" class="section level2">
<h2>Get Study Metadata</h2>
<p>Now that we are successfully connected, we may want to view all
studies available for our chosen database to find the correct
<code>study_id</code> corresponding to the data we want to pull. All
studies have a unique identifier in the database. You can view all
studies available in your database with the following:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>all_studies <span class="ot">&lt;-</span> <span class="fu">available_studies</span>()</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>all_studies</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a><span class="co">#&gt; # A tibble: 468 × 13</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a><span class="co">#&gt;    studyId          name    description publicStudy pmid  citation groups status importDate allSampleCount</span></span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;            &lt;chr&gt;   &lt;chr&gt;       &lt;lgl&gt;       &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt;   &lt;int&gt; &lt;chr&gt;               &lt;int&gt;</span></span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a><span class="co">#&gt;  1 acyc_mskcc_2013  Adenoi… Whole-exom… TRUE        2368… Ho et a… &quot;ACYC…      0 2023-12-0…             60</span></span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a><span class="co">#&gt;  2 acyc_fmi_2014    Adenoi… Targeted S… TRUE        2441… Ross et… &quot;ACYC…      0 2023-12-0…             28</span></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a><span class="co">#&gt;  3 acyc_jhu_2016    Adenoi… Whole-geno… TRUE        2686… Rettig … &quot;ACYC…      0 2023-12-0…             25</span></span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="co">#&gt;  4 acyc_mda_2015    Adenoi… WGS of 21 … TRUE        2663… Mitani … &quot;ACYC…      0 2023-12-0…            102</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="co">#&gt;  5 acyc_mgh_2016    Adenoi… Whole-geno… TRUE        2682… Drier e… &quot;ACYC&quot;      0 2023-12-0…             10</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="co">#&gt;  6 acyc_sanger_2013 Adenoi… Whole exom… TRUE        2377… Stephen… &quot;ACYC…      0 2023-12-0…             24</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="co">#&gt;  7 bcc_unige_2016   Basal … Whole-exom… TRUE        2695… Bonilla… &quot;PUBL…      0 2023-12-0…            293</span></span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="co">#&gt;  8 all_stjude_2015  Acute … Comprehens… TRUE        2573… Anderss… &quot;PUBL…      0 2023-12-0…             93</span></span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="co">#&gt;  9 ampca_bcm_2016   Ampull… Exome sequ… TRUE        2680… Gingras… &quot;PUBL…      0 2023-12-0…            160</span></span>
<span id="cb4-15"><a href="#cb4-15" tabindex="-1"></a><span class="co">#&gt; 10 all_stjude_2013  Hypodi… Whole geno… TRUE        2333… Holmfel… &quot;&quot;          0 2023-12-0…             44</span></span>
<span id="cb4-16"><a href="#cb4-16" tabindex="-1"></a><span class="co">#&gt; # ℹ 458 more rows</span></span>
<span id="cb4-17"><a href="#cb4-17" tabindex="-1"></a><span class="co">#&gt; # ℹ 3 more variables: readPermission &lt;lgl&gt;, cancerTypeId &lt;chr&gt;, referenceGenome &lt;chr&gt;</span></span></code></pre></div>
<p>By inspecting this data frame, we see the unique
<code>study_id</code> for the NMIBC data set is
<code>&quot;blca_nmibc_2017&quot;</code> and the unique <code>study_id</code> for
the prostate cancer data set is <code>&quot;prad_msk_2019&quot;</code>. To get
more information on our studies we can do the following:</p>
<p><em>Note: the transpose function <code>t()</code> is just used here
to better view results</em></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>all_studies <span class="sc">%&gt;%</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="fu">filter</span>(studyId <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;blca_nmibc_2017&quot;</span>, <span class="st">&quot;prad_msk_2019&quot;</span>))</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a><span class="co">#&gt; # A tibble: 2 × 13</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="co">#&gt;   studyId         name      description publicStudy pmid  citation groups status importDate allSampleCount</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;           &lt;chr&gt;     &lt;chr&gt;       &lt;lgl&gt;       &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt;   &lt;int&gt; &lt;chr&gt;               &lt;int&gt;</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="co">#&gt; 1 blca_nmibc_2017 Nonmuscl… IMPACT seq… TRUE        2858… Pietzak… PUBLIC      0 2023-12-0…            105</span></span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="co">#&gt; 2 prad_msk_2019   Prostate… MSK-IMPACT… TRUE        3156… Granlun… PUBLIC      0 2023-12-1…             18</span></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co">#&gt; # ℹ 3 more variables: readPermission &lt;lgl&gt;, cancerTypeId &lt;chr&gt;, referenceGenome &lt;chr&gt;</span></span></code></pre></div>
<p>More in-depth information about the study can be found with
<code>get_study_info()</code></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">get_study_info</span>(<span class="st">&quot;blca_nmibc_2017&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="co">#&gt;                             [,1]                                                                           </span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="co">#&gt; name                        &quot;Nonmuscle Invasive Bladder Cancer (MSK Eur Urol 2017)&quot;                        </span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="co">#&gt; description                 &quot;IMPACT sequencing of 105 High Risk Nonmuscle Invasive Bladder Cancer samples.&quot;</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="co">#&gt; publicStudy                 &quot;TRUE&quot;                                                                         </span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="co">#&gt; pmid                        &quot;28583311&quot;                                                                     </span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="co">#&gt; citation                    &quot;Pietzak et al. Eur Urol 2017&quot;                                                 </span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="co">#&gt; groups                      &quot;PUBLIC&quot;                                                                       </span></span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="co">#&gt; status                      &quot;0&quot;                                                                            </span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co">#&gt; importDate                  &quot;2023-12-07 10:15:31&quot;                                                          </span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="co">#&gt; allSampleCount              &quot;105&quot;                                                                          </span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="co">#&gt; sequencedSampleCount        &quot;105&quot;                                                                          </span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="co">#&gt; cnaSampleCount              &quot;105&quot;                                                                          </span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="co">#&gt; mrnaRnaSeqSampleCount       &quot;0&quot;                                                                            </span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a><span class="co">#&gt; mrnaRnaSeqV2SampleCount     &quot;0&quot;                                                                            </span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a><span class="co">#&gt; mrnaMicroarraySampleCount   &quot;0&quot;                                                                            </span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a><span class="co">#&gt; miRnaSampleCount            &quot;0&quot;                                                                            </span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a><span class="co">#&gt; methylationHm27SampleCount  &quot;0&quot;                                                                            </span></span>
<span id="cb6-20"><a href="#cb6-20" tabindex="-1"></a><span class="co">#&gt; rppaSampleCount             &quot;0&quot;                                                                            </span></span>
<span id="cb6-21"><a href="#cb6-21" tabindex="-1"></a><span class="co">#&gt; massSpectrometrySampleCount &quot;0&quot;                                                                            </span></span>
<span id="cb6-22"><a href="#cb6-22" tabindex="-1"></a><span class="co">#&gt; completeSampleCount         &quot;0&quot;                                                                            </span></span>
<span id="cb6-23"><a href="#cb6-23" tabindex="-1"></a><span class="co">#&gt; readPermission              &quot;TRUE&quot;                                                                         </span></span>
<span id="cb6-24"><a href="#cb6-24" tabindex="-1"></a><span class="co">#&gt; treatmentCount              &quot;0&quot;                                                                            </span></span>
<span id="cb6-25"><a href="#cb6-25" tabindex="-1"></a><span class="co">#&gt; structuralVariantCount      &quot;11&quot;                                                                           </span></span>
<span id="cb6-26"><a href="#cb6-26" tabindex="-1"></a><span class="co">#&gt; studyId                     &quot;blca_nmibc_2017&quot;                                                              </span></span>
<span id="cb6-27"><a href="#cb6-27" tabindex="-1"></a><span class="co">#&gt; cancerTypeId                &quot;blca&quot;                                                                         </span></span>
<span id="cb6-28"><a href="#cb6-28" tabindex="-1"></a><span class="co">#&gt; cancerType.name             &quot;Bladder Urothelial Carcinoma&quot;                                                 </span></span>
<span id="cb6-29"><a href="#cb6-29" tabindex="-1"></a><span class="co">#&gt; cancerType.dedicatedColor   &quot;Yellow&quot;                                                                       </span></span>
<span id="cb6-30"><a href="#cb6-30" tabindex="-1"></a><span class="co">#&gt; cancerType.shortName        &quot;BLCA&quot;                                                                         </span></span>
<span id="cb6-31"><a href="#cb6-31" tabindex="-1"></a><span class="co">#&gt; cancerType.parent           &quot;bladder&quot;                                                                      </span></span>
<span id="cb6-32"><a href="#cb6-32" tabindex="-1"></a><span class="co">#&gt; cancerType.cancerTypeId     &quot;blca&quot;                                                                         </span></span>
<span id="cb6-33"><a href="#cb6-33" tabindex="-1"></a><span class="co">#&gt; referenceGenome             &quot;hg19&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">get_study_info</span>(<span class="st">&quot;prad_msk_2019&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a><span class="co">#&gt;                             [,1]                                                             </span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a><span class="co">#&gt; name                        &quot;Prostate Cancer (MSK, Cell Metab 2020)&quot;                         </span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a><span class="co">#&gt; description                 &quot;MSK-IMPACT Sequencing of 18 prostate cancer tumor/normal pairs.&quot;</span></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co">#&gt; publicStudy                 &quot;TRUE&quot;                                                           </span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a><span class="co">#&gt; pmid                        &quot;31564440&quot;                                                       </span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a><span class="co">#&gt; citation                    &quot;Granlund et al. Cell Metab 2020&quot;                                </span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="co">#&gt; groups                      &quot;PUBLIC&quot;                                                         </span></span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a><span class="co">#&gt; status                      &quot;0&quot;                                                              </span></span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a><span class="co">#&gt; importDate                  &quot;2023-12-11 10:54:56&quot;                                            </span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a><span class="co">#&gt; allSampleCount              &quot;18&quot;                                                             </span></span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a><span class="co">#&gt; sequencedSampleCount        &quot;18&quot;                                                             </span></span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a><span class="co">#&gt; cnaSampleCount              &quot;18&quot;                                                             </span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a><span class="co">#&gt; mrnaRnaSeqSampleCount       &quot;0&quot;                                                              </span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a><span class="co">#&gt; mrnaRnaSeqV2SampleCount     &quot;0&quot;                                                              </span></span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a><span class="co">#&gt; mrnaMicroarraySampleCount   &quot;0&quot;                                                              </span></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a><span class="co">#&gt; miRnaSampleCount            &quot;0&quot;                                                              </span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a><span class="co">#&gt; methylationHm27SampleCount  &quot;0&quot;                                                              </span></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a><span class="co">#&gt; rppaSampleCount             &quot;0&quot;                                                              </span></span>
<span id="cb7-21"><a href="#cb7-21" tabindex="-1"></a><span class="co">#&gt; massSpectrometrySampleCount &quot;0&quot;                                                              </span></span>
<span id="cb7-22"><a href="#cb7-22" tabindex="-1"></a><span class="co">#&gt; completeSampleCount         &quot;0&quot;                                                              </span></span>
<span id="cb7-23"><a href="#cb7-23" tabindex="-1"></a><span class="co">#&gt; readPermission              &quot;TRUE&quot;                                                           </span></span>
<span id="cb7-24"><a href="#cb7-24" tabindex="-1"></a><span class="co">#&gt; treatmentCount              &quot;0&quot;                                                              </span></span>
<span id="cb7-25"><a href="#cb7-25" tabindex="-1"></a><span class="co">#&gt; structuralVariantCount      &quot;4&quot;                                                              </span></span>
<span id="cb7-26"><a href="#cb7-26" tabindex="-1"></a><span class="co">#&gt; studyId                     &quot;prad_msk_2019&quot;                                                  </span></span>
<span id="cb7-27"><a href="#cb7-27" tabindex="-1"></a><span class="co">#&gt; cancerTypeId                &quot;prostate&quot;                                                       </span></span>
<span id="cb7-28"><a href="#cb7-28" tabindex="-1"></a><span class="co">#&gt; cancerType.name             &quot;Prostate&quot;                                                       </span></span>
<span id="cb7-29"><a href="#cb7-29" tabindex="-1"></a><span class="co">#&gt; cancerType.dedicatedColor   &quot;Cyan&quot;                                                           </span></span>
<span id="cb7-30"><a href="#cb7-30" tabindex="-1"></a><span class="co">#&gt; cancerType.shortName        &quot;PROSTATE&quot;                                                       </span></span>
<span id="cb7-31"><a href="#cb7-31" tabindex="-1"></a><span class="co">#&gt; cancerType.parent           &quot;tissue&quot;                                                         </span></span>
<span id="cb7-32"><a href="#cb7-32" tabindex="-1"></a><span class="co">#&gt; cancerType.cancerTypeId     &quot;prostate&quot;                                                       </span></span>
<span id="cb7-33"><a href="#cb7-33" tabindex="-1"></a><span class="co">#&gt; referenceGenome             &quot;hg19&quot;</span></span></code></pre></div>
<p>Lastly, it is important to know what genomic data is available for
our studies. Not all studies in your database will have data available
on all types of genomic information. For example, it is common for
studies not to provide data on fusions/structural variants.</p>
<p>We can check available genomic data with
<code>available_profiles()</code>.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="fu">available_profiles</span>(<span class="at">study_id =</span> <span class="st">&quot;blca_nmibc_2017&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 8</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a><span class="co">#&gt;   molecularAlterationT…¹ datatype name  description showProfileInAnalysi…² patientLevel molecularProfileId</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                  &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;       &lt;lgl&gt;                  &lt;lgl&gt;        &lt;chr&gt;             </span></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a><span class="co">#&gt; 1 COPY_NUMBER_ALTERATION DISCRETE Puta… Copy Numbe… TRUE                   FALSE        blca_nmibc_2017_c…</span></span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="co">#&gt; 2 MUTATION_EXTENDED      MAF      Muta… Mutation d… TRUE                   FALSE        blca_nmibc_2017_m…</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="co">#&gt; 3 STRUCTURAL_VARIANT     SV       Stru… Structural… TRUE                   FALSE        blca_nmibc_2017_s…</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co">#&gt; # ℹ abbreviated names: ¹​molecularAlterationType, ²​showProfileInAnalysisTab</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="co">#&gt; # ℹ 1 more variable: studyId &lt;chr&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">available_profiles</span>(<span class="at">study_id =</span> <span class="st">&quot;prad_msk_2019&quot;</span>)</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="co">#&gt; # A tibble: 3 × 8</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="co">#&gt;   molecularAlterationT…¹ datatype name  description showProfileInAnalysi…² patientLevel molecularProfileId</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                  &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;       &lt;lgl&gt;                  &lt;lgl&gt;        &lt;chr&gt;             </span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="co">#&gt; 1 COPY_NUMBER_ALTERATION DISCRETE Puta… Putative c… TRUE                   FALSE        prad_msk_2019_cna </span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="co">#&gt; 2 MUTATION_EXTENDED      MAF      Muta… IMPACT468 … TRUE                   FALSE        prad_msk_2019_mut…</span></span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="co">#&gt; 3 STRUCTURAL_VARIANT     SV       Stru… Structural… TRUE                   FALSE        prad_msk_2019_str…</span></span>
<span id="cb9-8"><a href="#cb9-8" tabindex="-1"></a><span class="co">#&gt; # ℹ abbreviated names: ¹​molecularAlterationType, ²​showProfileInAnalysisTab</span></span>
<span id="cb9-9"><a href="#cb9-9" tabindex="-1"></a><span class="co">#&gt; # ℹ 1 more variable: studyId &lt;chr&gt;</span></span></code></pre></div>
<p>Luckily, in this example our studies have mutation, copy number
alteration and fusion (structural variant) data available. Each of these
data types has a unique molecular profile ID. The molecular profile ID
usually takes the form of <code>&lt;study_id&gt;_mutations</code>,
<code>&lt;study_id&gt;_structural_variants</code>,
<code>&lt;study_id&gt;_cna</code>.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">available_profiles</span>(<span class="at">study_id =</span> <span class="st">&quot;blca_nmibc_2017&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="fu">pull</span>(molecularProfileId)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="co">#&gt; [1] &quot;blca_nmibc_2017_cna&quot;                 &quot;blca_nmibc_2017_mutations&quot;          </span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="co">#&gt; [3] &quot;blca_nmibc_2017_structural_variants&quot;</span></span></code></pre></div>
</div>
<div id="pulling-genomic-data" class="section level2">
<h2>Pulling Genomic Data</h2>
<p>Now that we have inspected our studies and confirmed the genomic data
that is available, we will pull the data into our R environment. We will
show two ways to do this:</p>
<ol style="list-style-type: decimal">
<li>Using study IDs (<code>get_genetics_by_study()</code>)</li>
<li>Using sample ID-study ID pairs
(<code>get_genetics_by_sample()</code>)</li>
</ol>
<p>Pulling by study will give us genomic data for all genes/panels
included in the study. These functions can only pull data one study ID
at a time and will return all genomic data available for that study.
Pulling by study ID can be efficient, and a good way to ensure you have
all genomic information available in cBioPortal for a particular
study.</p>
<p>If you are working across multiple studies, or only need a subset of
samples from one or multiple studies, you may chose to pull by sample
IDs instead of study ID. When you pull by sample IDs you can pull
specific samples across multiple studies, but must also specify the
studies they belong to. You may also pass a specific list of genes for
which to return information. If you don’t specify a list of genes the
function will default to returning all available gene data for each
sample.</p>
<div id="by-study-ids" class="section level3">
<h3>By Study IDs</h3>
<p>To pull by study ID, we can pull each data type individually.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>mut_blca <span class="ot">&lt;-</span> <span class="fu">get_mutations_by_study</span>(<span class="at">study_id =</span> <span class="st">&quot;blca_nmibc_2017&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;blca_nmibc_2017_mutations&quot; molecular profile in the &quot;blca_nmibc_2017&quot; study</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>cna_blca<span class="ot">&lt;-</span> <span class="fu">get_cna_by_study</span>(<span class="at">study_id =</span> <span class="st">&quot;blca_nmibc_2017&quot;</span>)</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;blca_nmibc_2017_cna&quot; molecular profile in the &quot;blca_nmibc_2017&quot; study</span></span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a>fus_blca <span class="ot">&lt;-</span> <span class="fu">get_fusions_by_study</span>(<span class="at">study_id =</span> <span class="st">&quot;blca_nmibc_2017&quot;</span>)</span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;blca_nmibc_2017_structural_variants&quot; molecular profile in the &quot;blca_nmibc_2017&quot; study</span></span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>mut_prad <span class="ot">&lt;-</span> <span class="fu">get_mutations_by_study</span>(<span class="at">study_id =</span> <span class="st">&quot;prad_msk_2019&quot;</span>)</span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;prad_msk_2019_mutations&quot; molecular profile in the &quot;prad_msk_2019&quot; study</span></span>
<span id="cb11-12"><a href="#cb11-12" tabindex="-1"></a>cna_prad <span class="ot">&lt;-</span> <span class="fu">get_cna_by_study</span>(<span class="at">study_id =</span> <span class="st">&quot;prad_msk_2019&quot;</span>)</span>
<span id="cb11-13"><a href="#cb11-13" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;prad_msk_2019_cna&quot; molecular profile in the &quot;prad_msk_2019&quot; study</span></span>
<span id="cb11-14"><a href="#cb11-14" tabindex="-1"></a>fus_prad <span class="ot">&lt;-</span> <span class="fu">get_fusions_by_study</span>(<span class="at">study_id =</span> <span class="st">&quot;prad_msk_2019&quot;</span>)</span>
<span id="cb11-15"><a href="#cb11-15" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;prad_msk_2019_structural_variants&quot; molecular profile in the &quot;prad_msk_2019&quot; study</span></span></code></pre></div>
<p>Or we can pull all genomic data at the same time with
<code>get_genetics_by_study()</code></p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>all_genomic_blca <span class="ot">&lt;-</span> <span class="fu">get_genetics_by_study</span>(<span class="st">&quot;blca_nmibc_2017&quot;</span>)</span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;blca_nmibc_2017_mutations&quot; molecular profile in the &quot;blca_nmibc_2017&quot; study</span></span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;blca_nmibc_2017_cna&quot; molecular profile in the &quot;blca_nmibc_2017&quot; study</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;blca_nmibc_2017_structural_variants&quot; molecular profile in the &quot;blca_nmibc_2017&quot; study</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>all_genomic_prad <span class="ot">&lt;-</span> <span class="fu">get_genetics_by_study</span>(<span class="st">&quot;prad_msk_2019&quot;</span>)</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;prad_msk_2019_mutations&quot; molecular profile in the &quot;prad_msk_2019&quot; study</span></span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;prad_msk_2019_cna&quot; molecular profile in the &quot;prad_msk_2019&quot; study</span></span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;prad_msk_2019_structural_variants&quot; molecular profile in the &quot;prad_msk_2019&quot; study</span></span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">all_equal</span>(mut_blca, all_genomic_blca<span class="sc">$</span>mutation)</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a><span class="fu">all_equal</span>(cna_blca, all_genomic_blca<span class="sc">$</span>cna)</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span>
<span id="cb13-5"><a href="#cb13-5" tabindex="-1"></a><span class="fu">all_equal</span>(fus_blca, all_genomic_blca<span class="sc">$</span>structural_variant)</span>
<span id="cb13-6"><a href="#cb13-6" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>Finally, we can join the two studies together</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a>mut_study <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(mut_blca, mut_prad)</span>
<span id="cb14-2"><a href="#cb14-2" tabindex="-1"></a>cna_study <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cna_blca, cna_prad)</span>
<span id="cb14-3"><a href="#cb14-3" tabindex="-1"></a>fus_study <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(fus_blca, fus_prad)</span></code></pre></div>
</div>
<div id="by-sample-ids" class="section level3">
<h3>By Sample IDs</h3>
<p>When we pull by sample IDs, we can pull specific samples across
multiple studies. In the above example, we can pull from both studies at
the same time for a select set of samples using the
<code>sample_study_pairs</code> argument in
<code>get_genetics_by_sample()</code>.</p>
<p>Let’s pull data for the first 10 samples in each study. We first need
to construct our dataframe to pass to the function:</p>
<p><em>Note: you can also run
<code>available_samples(sample_list_id = &lt;sample list ID&gt;)</code>
to pull sample IDs by a specific sample list ID (see
<code>available_sample_lists()</code>), or
<code>available_patients()</code> to pull patient IDs</em></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>s1 <span class="ot">&lt;-</span> <span class="fu">available_samples</span>(<span class="st">&quot;blca_nmibc_2017&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>  <span class="fu">select</span>(sampleId, patientId, studyId) <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>)</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>s2 <span class="ot">&lt;-</span> <span class="fu">available_samples</span>(<span class="st">&quot;prad_msk_2019&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>  <span class="fu">select</span>(sampleId,  patientId, studyId) <span class="sc">%&gt;%</span></span>
<span id="cb15-7"><a href="#cb15-7" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">10</span>)</span>
<span id="cb15-8"><a href="#cb15-8" tabindex="-1"></a></span>
<span id="cb15-9"><a href="#cb15-9" tabindex="-1"></a>df_pairs <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(s1, s2) <span class="sc">%&gt;%</span></span>
<span id="cb15-10"><a href="#cb15-10" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>patientId)</span></code></pre></div>
<p>We need to rename the columns as per the functions documentation.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>df_pairs <span class="ot">&lt;-</span> df_pairs <span class="sc">%&gt;%</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;sample_id&quot;</span> <span class="ot">=</span> sampleId,</span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>         <span class="st">&quot;study_id&quot;</span> <span class="ot">=</span> studyId)</span></code></pre></div>
<p>Now we pass this to <code>get_genetics_by_sample()</code></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>all_genomic <span class="ot">&lt;-</span> <span class="fu">get_genetics_by_sample</span>(<span class="at">sample_study_pairs =</span> df_pairs)</span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(study_id)`</span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a><span class="co">#&gt; The following parameters were used in query:</span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a><span class="co">#&gt; Study ID: &quot;blca_nmibc_2017&quot; and &quot;prad_msk_2019&quot;</span></span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a><span class="co">#&gt; Molecular Profile ID: blca_nmibc_2017_mutations and prad_msk_2019_mutations</span></span>
<span id="cb17-6"><a href="#cb17-6" tabindex="-1"></a><span class="co">#&gt; Genes: &quot;All available genes&quot;</span></span>
<span id="cb17-7"><a href="#cb17-7" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(study_id)`</span></span>
<span id="cb17-8"><a href="#cb17-8" tabindex="-1"></a><span class="co">#&gt; The following parameters were used in query:</span></span>
<span id="cb17-9"><a href="#cb17-9" tabindex="-1"></a><span class="co">#&gt; Study ID: &quot;blca_nmibc_2017&quot; and &quot;prad_msk_2019&quot;</span></span>
<span id="cb17-10"><a href="#cb17-10" tabindex="-1"></a><span class="co">#&gt; Molecular Profile ID: blca_nmibc_2017_cna and prad_msk_2019_cna</span></span>
<span id="cb17-11"><a href="#cb17-11" tabindex="-1"></a><span class="co">#&gt; Genes: &quot;All available genes&quot;</span></span>
<span id="cb17-12"><a href="#cb17-12" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(study_id)`</span></span>
<span id="cb17-13"><a href="#cb17-13" tabindex="-1"></a><span class="co">#&gt; The following parameters were used in query:</span></span>
<span id="cb17-14"><a href="#cb17-14" tabindex="-1"></a><span class="co">#&gt; Study ID: &quot;blca_nmibc_2017&quot; and &quot;prad_msk_2019&quot;</span></span>
<span id="cb17-15"><a href="#cb17-15" tabindex="-1"></a><span class="co">#&gt; Molecular Profile ID: blca_nmibc_2017_structural_variants and prad_msk_2019_structural_variants</span></span>
<span id="cb17-16"><a href="#cb17-16" tabindex="-1"></a><span class="co">#&gt; Genes: &quot;All available genes&quot;</span></span>
<span id="cb17-17"><a href="#cb17-17" tabindex="-1"></a></span>
<span id="cb17-18"><a href="#cb17-18" tabindex="-1"></a>mut_sample <span class="ot">&lt;-</span> all_genomic<span class="sc">$</span>mutation</span></code></pre></div>
<p>Like with querying by study ID, you can also pull data individually
by genomic data type:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>mut_only <span class="ot">&lt;-</span> <span class="fu">get_mutations_by_sample</span>(<span class="at">sample_study_pairs =</span> df_pairs)</span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(study_id)`</span></span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a><span class="co">#&gt; The following parameters were used in query:</span></span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a><span class="co">#&gt; Study ID: &quot;blca_nmibc_2017&quot; and &quot;prad_msk_2019&quot;</span></span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a><span class="co">#&gt; Molecular Profile ID: blca_nmibc_2017_mutations and prad_msk_2019_mutations</span></span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a><span class="co">#&gt; Genes: &quot;All available genes&quot;</span></span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a></span>
<span id="cb18-8"><a href="#cb18-8" tabindex="-1"></a><span class="fu">identical</span>(mut_only, mut_sample)</span>
<span id="cb18-9"><a href="#cb18-9" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>Let’s compare these results with the ones we got from pulling by
study:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a><span class="co"># filter to our subset used in sample query</span></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>mut_study_subset <span class="ot">&lt;-</span> mut_study <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>  <span class="fu">filter</span>(sampleId <span class="sc">%in%</span>  df_pairs<span class="sc">$</span>sample_id)</span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a></span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a><span class="co"># arrange to compare</span></span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>mut_study_subset <span class="ot">&lt;-</span> mut_study_subset <span class="sc">%&gt;%</span></span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(sampleId))<span class="sc">%&gt;%</span></span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(entrezGeneId))</span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a></span>
<span id="cb19-11"><a href="#cb19-11" tabindex="-1"></a>mut_sample <span class="ot">&lt;-</span> mut_sample <span class="sc">%&gt;%</span></span>
<span id="cb19-12"><a href="#cb19-12" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(sampleId)) <span class="sc">%&gt;%</span></span>
<span id="cb19-13"><a href="#cb19-13" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(entrezGeneId)) <span class="sc">%&gt;%</span></span>
<span id="cb19-14"><a href="#cb19-14" tabindex="-1"></a></span>
<span id="cb19-15"><a href="#cb19-15" tabindex="-1"></a>  <span class="co"># reorder so columns in same order</span></span>
<span id="cb19-16"><a href="#cb19-16" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">names</span>(mut_study_subset))</span>
<span id="cb19-17"><a href="#cb19-17" tabindex="-1"></a></span>
<span id="cb19-18"><a href="#cb19-18" tabindex="-1"></a><span class="fu">all.equal</span>(mut_study_subset, mut_sample)</span>
<span id="cb19-19"><a href="#cb19-19" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>Both results are equal.</p>
<p>Note: some studies also have copy number segments data available that
can be pulled by study ID or sample ID:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>seg_blca <span class="ot">&lt;-</span> <span class="fu">get_segments_by_study</span>(<span class="st">&quot;blca_nmibc_2017&quot;</span>)</span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all &quot;copy number segmentation&quot; data for the &quot;blca_nmibc_2017&quot; study</span></span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a></span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a><span class="co"># To pull alongside other genomic data types, use the `return_segments` argument</span></span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>all_genomic_blca <span class="ot">&lt;-</span> <span class="fu">get_genetics_by_study</span>(<span class="st">&quot;blca_nmibc_2017&quot;</span>, <span class="at">return_segments =</span> <span class="cn">TRUE</span>)</span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;blca_nmibc_2017_mutations&quot; molecular profile in the &quot;blca_nmibc_2017&quot; study</span></span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;blca_nmibc_2017_cna&quot; molecular profile in the &quot;blca_nmibc_2017&quot; study</span></span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all data for the &quot;blca_nmibc_2017_structural_variants&quot; molecular profile in the &quot;blca_nmibc_2017&quot; study</span></span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a><span class="co">#&gt; ℹ Returning all &quot;copy number segmentation&quot; data for the &quot;blca_nmibc_2017&quot; study</span></span></code></pre></div>
<div id="limit-results-to-specified-genes-or-panels" class="section level4">
<h4>Limit Results to Specified Genes or Panels</h4>
<p>When pulling by sample IDs, we can also limit our results to a
specific set of genes by passing a vector of Entrez Gene IDs or Hugo
Symbols to the <code>gene</code> argument, or a specified panel by
passing a panel ID to the <code>panel</code> argument (see
<code>available_gene_panels()</code> for supported panels). This can be
useful if, for example, we want to pull all IMPACT gene results for two
studies but one of the two uses a much larger panel. In that case, we
can limit our query to just the genes for which we want results:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a>by_hugo <span class="ot">&lt;-</span> <span class="fu">get_mutations_by_sample</span>(<span class="at">sample_study_pairs =</span> df_pairs, <span class="at">genes =</span> <span class="st">&quot;TP53&quot;</span>)</span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(study_id)`</span></span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a><span class="co">#&gt; The following parameters were used in query:</span></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a><span class="co">#&gt; Study ID: &quot;blca_nmibc_2017&quot; and &quot;prad_msk_2019&quot;</span></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a><span class="co">#&gt; Molecular Profile ID: blca_nmibc_2017_mutations and prad_msk_2019_mutations</span></span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a><span class="co">#&gt; Genes: &quot;TP53&quot;</span></span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a>by_gene_id <span class="ot">&lt;-</span> <span class="fu">get_mutations_by_sample</span>(<span class="at">sample_study_pairs =</span> df_pairs, <span class="at">genes =</span> <span class="dv">7157</span>)</span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(study_id)`</span></span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a><span class="co">#&gt; The following parameters were used in query:</span></span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a><span class="co">#&gt; Study ID: &quot;blca_nmibc_2017&quot; and &quot;prad_msk_2019&quot;</span></span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a><span class="co">#&gt; Molecular Profile ID: blca_nmibc_2017_mutations and prad_msk_2019_mutations</span></span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a><span class="co">#&gt; Genes: 7157</span></span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a></span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a><span class="fu">identical</span>(by_hugo, by_gene_id)</span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">get_mutations_by_sample</span>(</span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>  <span class="at">sample_study_pairs =</span> df_pairs,</span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a>  <span class="at">panel =</span> <span class="st">&quot;IMPACT468&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>  <span class="fu">head</span>()</span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a><span class="co">#&gt; Joining with `by = join_by(study_id)`</span></span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a><span class="co">#&gt; The following parameters were used in query:</span></span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a><span class="co">#&gt; Study ID: &quot;blca_nmibc_2017&quot; and &quot;prad_msk_2019&quot;</span></span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a><span class="co">#&gt; Molecular Profile ID: blca_nmibc_2017_mutations and prad_msk_2019_mutations</span></span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a><span class="co">#&gt; Genes: &quot;IMPACT468&quot;</span></span>
<span id="cb22-10"><a href="#cb22-10" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 × 28</span></span>
<span id="cb22-11"><a href="#cb22-11" tabindex="-1"></a><span class="co">#&gt;   hugoGeneSymbol entrezGeneId uniqueSampleKey       uniquePatientKey molecularProfileId sampleId patientId</span></span>
<span id="cb22-12"><a href="#cb22-12" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;                 &lt;int&gt; &lt;chr&gt;                 &lt;chr&gt;            &lt;chr&gt;              &lt;chr&gt;    &lt;chr&gt;    </span></span>
<span id="cb22-13"><a href="#cb22-13" tabindex="-1"></a><span class="co">#&gt; 1 TERT                   7015 UC0wMDAxNDUzLVQwMS1J… UC0wMDAxNDUzOmJ… blca_nmibc_2017_m… P-00014… P-0001453</span></span>
<span id="cb22-14"><a href="#cb22-14" tabindex="-1"></a><span class="co">#&gt; 2 SMAD4                  4089 UC0wMDAxNDUzLVQwMS1J… UC0wMDAxNDUzOmJ… blca_nmibc_2017_m… P-00014… P-0001453</span></span>
<span id="cb22-15"><a href="#cb22-15" tabindex="-1"></a><span class="co">#&gt; 3 ERBB4                  2066 UC0wMDAxNDUzLVQwMS1J… UC0wMDAxNDUzOmJ… blca_nmibc_2017_m… P-00014… P-0001453</span></span>
<span id="cb22-16"><a href="#cb22-16" tabindex="-1"></a><span class="co">#&gt; 4 CUL3                   8452 UC0wMDAxNDUzLVQwMS1J… UC0wMDAxNDUzOmJ… blca_nmibc_2017_m… P-00014… P-0001453</span></span>
<span id="cb22-17"><a href="#cb22-17" tabindex="-1"></a><span class="co">#&gt; 5 PBRM1                 55193 UC0wMDAxNDUzLVQwMS1J… UC0wMDAxNDUzOmJ… blca_nmibc_2017_m… P-00014… P-0001453</span></span>
<span id="cb22-18"><a href="#cb22-18" tabindex="-1"></a><span class="co">#&gt; 6 APC                     324 UC0wMDAxNDUzLVQwMS1J… UC0wMDAxNDUzOmJ… blca_nmibc_2017_m… P-00014… P-0001453</span></span>
<span id="cb22-19"><a href="#cb22-19" tabindex="-1"></a><span class="co">#&gt; # ℹ 21 more variables: studyId &lt;chr&gt;, center &lt;chr&gt;, mutationStatus &lt;chr&gt;, validationStatus &lt;chr&gt;,</span></span>
<span id="cb22-20"><a href="#cb22-20" tabindex="-1"></a><span class="co">#&gt; #   tumorAltCount &lt;int&gt;, tumorRefCount &lt;int&gt;, normalAltCount &lt;int&gt;, normalRefCount &lt;int&gt;,</span></span>
<span id="cb22-21"><a href="#cb22-21" tabindex="-1"></a><span class="co">#&gt; #   startPosition &lt;int&gt;, endPosition &lt;int&gt;, referenceAllele &lt;chr&gt;, proteinChange &lt;chr&gt;,</span></span>
<span id="cb22-22"><a href="#cb22-22" tabindex="-1"></a><span class="co">#&gt; #   mutationType &lt;chr&gt;, ncbiBuild &lt;chr&gt;, variantType &lt;chr&gt;, chr &lt;chr&gt;, variantAllele &lt;chr&gt;,</span></span>
<span id="cb22-23"><a href="#cb22-23" tabindex="-1"></a><span class="co">#&gt; #   refseqMrnaId &lt;chr&gt;, proteinPosStart &lt;int&gt;, proteinPosEnd &lt;int&gt;, keyword &lt;chr&gt;</span></span></code></pre></div>
</div>
</div>
</div>
<div id="pulling-clinical-data-sample-metadata" class="section level2">
<h2>Pulling Clinical Data &amp; Sample Metadata</h2>
<p>You can also pull clinical data by study ID, sample ID, or patient
ID. Pulling by sample ID will pull all sample-level characteristics
(e.g. sample site, tumor stage at sampling time and other variables
collected at time of sampling that may be available). Pulling by patient
ID will pull all patient-level characteristics (e.g. age, sex, etc.).
Pulling by study ID will pull all sample <em>and</em> patient-level
characteristics at once.</p>
<p>You can explore what clinical data is available a study using:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a>attr_blca <span class="ot">&lt;-</span> <span class="fu">available_clinical_attributes</span>(<span class="st">&quot;blca_nmibc_2017&quot;</span>)</span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>attr_prad <span class="ot">&lt;-</span> <span class="fu">available_clinical_attributes</span>(<span class="st">&quot;prad_msk_2019&quot;</span>)</span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>attr_prad</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a><span class="co">#&gt; # A tibble: 13 × 7</span></span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a><span class="co">#&gt;    displayName                  description datatype patientAttribute priority clinicalAttributeId studyId</span></span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;                        &lt;chr&gt;       &lt;chr&gt;    &lt;lgl&gt;            &lt;chr&gt;    &lt;chr&gt;               &lt;chr&gt;  </span></span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a><span class="co">#&gt;  1 Cancer Type                  Cancer Type STRING   FALSE            1        CANCER_TYPE         prad_m…</span></span>
<span id="cb23-9"><a href="#cb23-9" tabindex="-1"></a><span class="co">#&gt;  2 Cancer Type Detailed         Cancer Typ… STRING   FALSE            1        CANCER_TYPE_DETAIL… prad_m…</span></span>
<span id="cb23-10"><a href="#cb23-10" tabindex="-1"></a><span class="co">#&gt;  3 Fraction Genome Altered      Fraction G… NUMBER   FALSE            20       FRACTION_GENOME_AL… prad_m…</span></span>
<span id="cb23-11"><a href="#cb23-11" tabindex="-1"></a><span class="co">#&gt;  4 Gene Panel                   Gene Panel. STRING   FALSE            1        GENE_PANEL          prad_m…</span></span>
<span id="cb23-12"><a href="#cb23-12" tabindex="-1"></a><span class="co">#&gt;  5 Mutation Count               Mutation C… NUMBER   FALSE            30       MUTATION_COUNT      prad_m…</span></span>
<span id="cb23-13"><a href="#cb23-13" tabindex="-1"></a><span class="co">#&gt;  6 Oncotree Code                Oncotree C… STRING   FALSE            1        ONCOTREE_CODE       prad_m…</span></span>
<span id="cb23-14"><a href="#cb23-14" tabindex="-1"></a><span class="co">#&gt;  7 Sample Class                 The sample… STRING   FALSE            1        SAMPLE_CLASS        prad_m…</span></span>
<span id="cb23-15"><a href="#cb23-15" tabindex="-1"></a><span class="co">#&gt;  8 Number of Samples Per Patie… Number of … STRING   TRUE             1        SAMPLE_COUNT        prad_m…</span></span>
<span id="cb23-16"><a href="#cb23-16" tabindex="-1"></a><span class="co">#&gt;  9 Sample Type                  The type o… STRING   FALSE            1        SAMPLE_TYPE         prad_m…</span></span>
<span id="cb23-17"><a href="#cb23-17" tabindex="-1"></a><span class="co">#&gt; 10 Sex                          Sex         STRING   TRUE             1        SEX                 prad_m…</span></span>
<span id="cb23-18"><a href="#cb23-18" tabindex="-1"></a><span class="co">#&gt; 11 Somatic Status               Somatic St… STRING   FALSE            1        SOMATIC_STATUS      prad_m…</span></span>
<span id="cb23-19"><a href="#cb23-19" tabindex="-1"></a><span class="co">#&gt; 12 Specimen Preservation Type   The method… STRING   FALSE            1        SPECIMEN_PRESERVAT… prad_m…</span></span>
<span id="cb23-20"><a href="#cb23-20" tabindex="-1"></a><span class="co">#&gt; 13 TMB (nonsynonymous)          TMB (nonsy… NUMBER   FALSE            1        TMB_NONSYNONYMOUS   prad_m…</span></span></code></pre></div>
<p>There are a select set available for both studies:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a>in_both <span class="ot">&lt;-</span> <span class="fu">intersect</span>(attr_blca<span class="sc">$</span>clinicalAttributeId, attr_prad<span class="sc">$</span>clinicalAttributeId)</span></code></pre></div>
<p>The below pulls data at the sample level:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a>clinical_blca <span class="ot">&lt;-</span> <span class="fu">get_clinical_by_sample</span>(<span class="at">sample_id =</span> s1<span class="sc">$</span>sampleId,</span>
<span id="cb25-2"><a href="#cb25-2" tabindex="-1"></a>                       <span class="at">study_id =</span> <span class="st">&quot;blca_nmibc_2017&quot;</span>,</span>
<span id="cb25-3"><a href="#cb25-3" tabindex="-1"></a>                       <span class="at">clinical_attribute =</span> in_both)</span>
<span id="cb25-4"><a href="#cb25-4" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" tabindex="-1"></a>clinical_prad <span class="ot">&lt;-</span> <span class="fu">get_clinical_by_sample</span>(<span class="at">sample_id =</span> s2<span class="sc">$</span>sampleId,</span>
<span id="cb25-6"><a href="#cb25-6" tabindex="-1"></a>                       <span class="at">study_id =</span> <span class="st">&quot;prad_msk_2019&quot;</span>,</span>
<span id="cb25-7"><a href="#cb25-7" tabindex="-1"></a>                       <span class="at">clinical_attribute =</span> in_both)</span>
<span id="cb25-8"><a href="#cb25-8" tabindex="-1"></a></span>
<span id="cb25-9"><a href="#cb25-9" tabindex="-1"></a>all_clinical <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(clinical_blca, clinical_prad)</span>
<span id="cb25-10"><a href="#cb25-10" tabindex="-1"></a></span>
<span id="cb25-11"><a href="#cb25-11" tabindex="-1"></a>all_clinical <span class="sc">%&gt;%</span></span>
<span id="cb25-12"><a href="#cb25-12" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">contains</span>(<span class="st">&quot;unique&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb25-13"><a href="#cb25-13" tabindex="-1"></a>  <span class="fu">head</span>()</span>
<span id="cb25-14"><a href="#cb25-14" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 × 5</span></span>
<span id="cb25-15"><a href="#cb25-15" tabindex="-1"></a><span class="co">#&gt;   sampleId          patientId studyId         clinicalAttributeId     value                       </span></span>
<span id="cb25-16"><a href="#cb25-16" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;             &lt;chr&gt;     &lt;chr&gt;           &lt;chr&gt;                   &lt;chr&gt;                       </span></span>
<span id="cb25-17"><a href="#cb25-17" tabindex="-1"></a><span class="co">#&gt; 1 P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 CANCER_TYPE             Bladder Cancer              </span></span>
<span id="cb25-18"><a href="#cb25-18" tabindex="-1"></a><span class="co">#&gt; 2 P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 CANCER_TYPE_DETAILED    Bladder Urothelial Carcinoma</span></span>
<span id="cb25-19"><a href="#cb25-19" tabindex="-1"></a><span class="co">#&gt; 3 P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 FRACTION_GENOME_ALTERED 0.4448                      </span></span>
<span id="cb25-20"><a href="#cb25-20" tabindex="-1"></a><span class="co">#&gt; 4 P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 MUTATION_COUNT          11                          </span></span>
<span id="cb25-21"><a href="#cb25-21" tabindex="-1"></a><span class="co">#&gt; 5 P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 ONCOTREE_CODE           BLCA                        </span></span>
<span id="cb25-22"><a href="#cb25-22" tabindex="-1"></a><span class="co">#&gt; 6 P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 SOMATIC_STATUS          Matched</span></span></code></pre></div>
<p>The below pulls data at the patient level:</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">available_patients</span>(<span class="st">&quot;blca_nmibc_2017&quot;</span>)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>clinical_blca <span class="ot">&lt;-</span> <span class="fu">get_clinical_by_patient</span>(<span class="at">patient_id =</span> s1<span class="sc">$</span>patientId,</span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>                       <span class="at">study_id =</span> <span class="st">&quot;blca_nmibc_2017&quot;</span>,</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>                       <span class="at">clinical_attribute =</span> in_both)</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" tabindex="-1"></a>clinical_prad <span class="ot">&lt;-</span> <span class="fu">get_clinical_by_patient</span>(<span class="at">patient_id =</span> s2<span class="sc">$</span>patientId,</span>
<span id="cb26-8"><a href="#cb26-8" tabindex="-1"></a>                       <span class="at">study_id =</span> <span class="st">&quot;prad_msk_2019&quot;</span>,</span>
<span id="cb26-9"><a href="#cb26-9" tabindex="-1"></a>                       <span class="at">clinical_attribute =</span> in_both)</span>
<span id="cb26-10"><a href="#cb26-10" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" tabindex="-1"></a>all_clinical <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(clinical_blca, clinical_prad)</span>
<span id="cb26-12"><a href="#cb26-12" tabindex="-1"></a></span>
<span id="cb26-13"><a href="#cb26-13" tabindex="-1"></a>all_clinical <span class="sc">%&gt;%</span></span>
<span id="cb26-14"><a href="#cb26-14" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">contains</span>(<span class="st">&quot;unique&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-15"><a href="#cb26-15" tabindex="-1"></a>  <span class="fu">head</span>()</span>
<span id="cb26-16"><a href="#cb26-16" tabindex="-1"></a><span class="co">#&gt; # A tibble: 6 × 4</span></span>
<span id="cb26-17"><a href="#cb26-17" tabindex="-1"></a><span class="co">#&gt;   patientId studyId         clinicalAttributeId value</span></span>
<span id="cb26-18"><a href="#cb26-18" tabindex="-1"></a><span class="co">#&gt;   &lt;chr&gt;     &lt;chr&gt;           &lt;chr&gt;               &lt;chr&gt;</span></span>
<span id="cb26-19"><a href="#cb26-19" tabindex="-1"></a><span class="co">#&gt; 1 P-0001453 blca_nmibc_2017 SAMPLE_COUNT        1    </span></span>
<span id="cb26-20"><a href="#cb26-20" tabindex="-1"></a><span class="co">#&gt; 2 P-0001453 blca_nmibc_2017 SEX                 Male </span></span>
<span id="cb26-21"><a href="#cb26-21" tabindex="-1"></a><span class="co">#&gt; 3 P-0002166 blca_nmibc_2017 SAMPLE_COUNT        1    </span></span>
<span id="cb26-22"><a href="#cb26-22" tabindex="-1"></a><span class="co">#&gt; 4 P-0002166 blca_nmibc_2017 SEX                 Male </span></span>
<span id="cb26-23"><a href="#cb26-23" tabindex="-1"></a><span class="co">#&gt; 5 P-0003238 blca_nmibc_2017 SAMPLE_COUNT        1    </span></span>
<span id="cb26-24"><a href="#cb26-24" tabindex="-1"></a><span class="co">#&gt; 6 P-0003238 blca_nmibc_2017 SEX                 Male</span></span></code></pre></div>
<p>Like with the genomic data pull functions, you can also pull clinical
data by a data frame of sample ID - study ID pairs, or a data frame of
patient ID - study ID pairs. Below, we will pull by patient ID - study
ID pairs.</p>
<p>First, we construct the data frame of pairs to pass:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>df_pairs <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(s1, s2) <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>sampleId)</span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>df_pairs <span class="ot">&lt;-</span> df_pairs <span class="sc">%&gt;%</span></span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a>  <span class="fu">select</span>(patientId, studyId)</span></code></pre></div>
<p>Now we pass this data frame to
<code>get_genetics_by_patient()</code></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>all_patient_clinical <span class="ot">&lt;-</span> <span class="fu">get_clinical_by_patient</span>(<span class="at">patient_study_pairs =</span> df_pairs,</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>                                                <span class="at">clinical_attribute =</span> in_both)</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>all_patient_clinical <span class="sc">%&gt;%</span></span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">contains</span>(<span class="st">&quot;unique&quot;</span>))</span>
<span id="cb28-6"><a href="#cb28-6" tabindex="-1"></a><span class="co">#&gt; # A tibble: 34 × 4</span></span>
<span id="cb28-7"><a href="#cb28-7" tabindex="-1"></a><span class="co">#&gt;    patientId studyId         clinicalAttributeId value </span></span>
<span id="cb28-8"><a href="#cb28-8" tabindex="-1"></a><span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;           &lt;chr&gt;               &lt;chr&gt; </span></span>
<span id="cb28-9"><a href="#cb28-9" tabindex="-1"></a><span class="co">#&gt;  1 P-0001453 blca_nmibc_2017 SAMPLE_COUNT        1     </span></span>
<span id="cb28-10"><a href="#cb28-10" tabindex="-1"></a><span class="co">#&gt;  2 P-0001453 blca_nmibc_2017 SEX                 Male  </span></span>
<span id="cb28-11"><a href="#cb28-11" tabindex="-1"></a><span class="co">#&gt;  3 P-0002166 blca_nmibc_2017 SAMPLE_COUNT        1     </span></span>
<span id="cb28-12"><a href="#cb28-12" tabindex="-1"></a><span class="co">#&gt;  4 P-0002166 blca_nmibc_2017 SEX                 Male  </span></span>
<span id="cb28-13"><a href="#cb28-13" tabindex="-1"></a><span class="co">#&gt;  5 P-0003238 blca_nmibc_2017 SAMPLE_COUNT        1     </span></span>
<span id="cb28-14"><a href="#cb28-14" tabindex="-1"></a><span class="co">#&gt;  6 P-0003238 blca_nmibc_2017 SEX                 Male  </span></span>
<span id="cb28-15"><a href="#cb28-15" tabindex="-1"></a><span class="co">#&gt;  7 P-0003257 blca_nmibc_2017 SAMPLE_COUNT        1     </span></span>
<span id="cb28-16"><a href="#cb28-16" tabindex="-1"></a><span class="co">#&gt;  8 P-0003257 blca_nmibc_2017 SEX                 Female</span></span>
<span id="cb28-17"><a href="#cb28-17" tabindex="-1"></a><span class="co">#&gt;  9 P-0003261 blca_nmibc_2017 SAMPLE_COUNT        1     </span></span>
<span id="cb28-18"><a href="#cb28-18" tabindex="-1"></a><span class="co">#&gt; 10 P-0003261 blca_nmibc_2017 SEX                 Male  </span></span>
<span id="cb28-19"><a href="#cb28-19" tabindex="-1"></a><span class="co">#&gt; # ℹ 24 more rows</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
